### DISCUSSION TITLE
Add a cli argument to close the running ui

### DISCUSSION THREAD

**Discussion by mattmart3** [abenz1267/walker#184] [Ideas]
What I am used to do with other launchers is to press the same keybindings to open and close the launcher (toggle). For example it's the default behavior on KDE with krunner (MOD+Space opens and closes the runner). With rofi I was achieving this by associating the keybinding to a command ike `pkill rofi || rofi`.  I can do the same with walker like `pkill walker || walker` but that would not keep the service application running if I launch walker with `--gapplication-service`. It looks like there is no cli argument to close a running UI instance. Would that be feasible? Is there another way to achieve this?

**Comment by abenz1267** [comment]
I've added a `close_when_open` boolean which defaults to `false`. This should achieve what you want.

**Reply by mattmart3** [reply]
That's great thanks! I'll test it at first chance.

**Reply by abenz1267** [reply]
Should be updated in the AUR.

**Reply by mattmart3** [reply]
Works perfectly, thanks!
